source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -764,6 +776,13 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   typed_tls_data.context_ = &json_body;
   typed_tls_data.environ_ = &environ_;
   typed_tls_data.cluster_metadata_ = cluster_metadata;
+  typed_tls_data.dynamic_metadata_ = dynamic_metadata;
+
+  // If there is a value we're in a upstream filter
+  if (callbacks.upstreamCallbacks().has_value()) {
+    auto &upstream_callbacks = callbacks.upstreamCallbacks().value().get();
+    typed_tls_data.endpoint_metadata_ = upstream_callbacks.upstreamStreamInfo().upstreamInfo()->upstreamHost()->metadata();
+  }
```

Comment:
check if `upstreamHost` is null here (in case we don't setup the wait filter in the control plane)

Reply:
Hmm, right now you we always use them together in the `upstream_context`, do we want to plan for it not being there if the control-plane never allows that situation?

Reply:
idk how we might change the control plane in the future.. i think it's better to verify here